ftp_blueprints/move_file.py

```diff
@@ -0,0 +1,207 @@
+        print(f"Ensure that the source/destination file name and folder name are correct")
+        sys.exit(ec.EXIT_CODE_FTP_MOVE_ERROR)
+
+    print(f'{source_path} successfully moved to '
+          f'{dest_path}')
+
+
+def get_client(host, port, username, password):
+    """
+    Attempts to create an FTP client at the specified hots with the
+    specified credentials
+    """
+    try:
+        client = ftplib.FTP()
+        # client.set_pasv(False) ## setting to active mode
+        client.connect(host, int(port))
+        client.login(username, password)
+        return client
+    except Exception as e:
+        print(f'Error accessing the FTP server with the specified credentials')
+        print(f'The server says: {e}')
+        sys.exit(ec.EXIT_CODE_INCORRECT_CREDENTIALS)
+
+
+def main():
+    args = get_args()
+    host = args.host
+    port = args.port
+    username = args.username
+    password = args.password
+    source_file_name = args.source_file_name
+    source_folder_name = args.source_folder_name
+    source_full_path = shipyard.files.combine_folder_and_file_name(
+        source_folder_name,
+        source_file_name)
+    destination_folder_name = shipyard.files.clean_folder_name(args.destination_folder_name)
+    source_file_name_match_type = args.source_file_name_match_type
+    client = get_client(host=host, port=port, username=username,
+                        password=password)
+    ## if the source folder name is omitted, then use the current working direcotry
+    if source_folder_name == '':
+        source_folder_name = client.pwd()
+    if source_file_name_match_type == 'regex_match':
+        file_names, folders = find_files_in_directory(client, source_folder_name)
+        matching_file_names = shipyard.files.find_all_file_matches(
+            file_names, re.compile(source_file_name))
+
+        number_of_matches = len(matching_file_names)
+
+        if number_of_matches == 0:
+            print(f'No matches were found for regex "{source_file_name}".')
+            sys.exit(ec.EXIT_CODE_NO_MATCHES_FOUND)
+
+        print(f'{len(matching_file_names)} files found. Preparing to move...')
+
+        for index, key_name in enumerate(matching_file_names,1):
+            destination_full_path = shipyard.files.determine_destination_full_path(
+                destination_folder_name = destination_folder_name,
+                destination_file_name = args.destination_file_name, 
+                source_full_path = key_name,
+                file_number = None if len(matching_file_names) == 1 else index
+
+            )
+            if len(destination_full_path.split('/')) > 1:
+                path, file_name = destination_full_path.rsplit('/', 1)
+                ftp_create_new_folders(client=client, destination_path=path)
+            file_name = destination_full_path.rsplit('/', 1)[-1]
+            print(f'Moving file {index+1} of {len(matching_file_names)}')
```

Comment:
No need for +1 here since you started at 1. It ends up saying "moving file 2 of 1".